pkg/cassandra/util.go

```diff
@@ -31,14 +33,38 @@ func DatacenterStopping(dc *cassdcapi.CassandraDatacenter) bool {
 	return dc.GetConditionStatus(cassdcapi.DatacenterStopped) == corev1.ConditionTrue && dc.Status.CassandraOperatorProgress == cassdcapi.ProgressUpdating
 }
 
-func ComputeSystemReplication(kluster *api.K8ssandraCluster) SystemReplication {
+// GetDatacentersForSystemReplication determines the DCs that should be included for
+// replication. This function should only be used for system keyspaces. Replication for
+// system keyspaces is initially set through the management-api, not CQL. This allows us
+// to specify non-existent DCs for replication even though Cassandra 4 does not allow that.
+// That cannot be done when configuration replication through CQL which is why this func
+// should only be used for system keyspaces.
+func GetDatacentersForSystemReplication(kc *api.K8ssandraCluster) []api.CassandraDatacenterTemplate {
+	var datacenters []api.CassandraDatacenterTemplate
+	if kc.Status.GetConditionStatus(api.CassandraInitialized) == corev1.ConditionTrue {
+		datacenters = make([]api.CassandraDatacenterTemplate, 0)
+		for _, dc := range kc.Spec.Cassandra.Datacenters {
+			if status, found := kc.Status.Datacenters[dc.Meta.Name]; found && status.Cassandra.GetConditionStatus(cassdcapi.DatacenterReady) == corev1.ConditionTrue {
+				datacenters = append(datacenters, dc)
+			}
+		}
+	} else {
+		datacenters = kc.Spec.Cassandra.Datacenters
+	}
+	return datacenters
+}
+
+func ComputeSystemReplication(kc *api.K8ssandraCluster) SystemReplication {
```

Comment:
I see that this function is now only called from `checkSystemReplication`, which means that this code is only executed for newly-created clusters, and the purpose here is then to compute the _initial_ system replication. Couldn't we rename this to `ComputeInitialSystemReplication` then?

Reply:
👍